packages/client/src/pages/app/AppDetailPage.tsx

```diff
@@ -239,105 +239,99 @@ export const AppDetailPage = () => {
             ),
             fetchMainUses(monolithStore, id),
-            fetchDependencies(monolithStore, id),
-        ]).then((results) =>
-            results.forEach((res, idx) => {
-                if (res.status === 'rejected') {
-                    emitMessage(true, res.reason);
-                } else {
-                    if (idx === 0) {
-                        if (res.value.type === 'error') {
-                            emitMessage(true, res.value.output);
-                        } else {
-                            setValue('appInfo', res.value.output);
-                            const output = res.value.output;
-
-                            const projectMetaKeys =
-                                configStore.store.config.projectMetaKeys;
-                            // Keep only relevant project keys defined for app details
-                            const parsedMeta = projectMetaKeys
-                                .map((k) => k.metakey)
-                                .reduce((prev, curr) => {
-                                    // tag, domain, and etc either come in as a string or a string[], format it to correct type
-                                    const found = projectMetaKeys.find(
-                                        (obj) => obj.metakey === curr,
-                                    );
-
-                                    if (curr === 'tag') {
-                                        if (typeof output[curr] === 'string') {
-                                            prev[curr] = [output[curr]];
-                                        } else {
-                                            prev[curr] = output[curr];
-                                        }
-                                    } else if (
-                                        found.display_options ===
-                                            'single-typeahead' ||
-                                        found.display_options ===
-                                            'select-box' ||
-                                        found.display_options ===
-                                            'multi-typeahead'
-                                    ) {
-                                        if (typeof output[curr] === 'string') {
-                                            prev[curr] = [output[curr]];
-                                        } else {
-                                            prev[curr] = output[curr];
-                                        }
+        ];
+        if (permission !== 'discoverable') {
+            promises.push(fetchDependencies(monolithStore, id));
+        }
+        const results = await Promise.allSettled(promises);
+        results.forEach((res, idx) => {
+            if (res.status === 'rejected') {
+                emitMessage(true, res.reason);
+            } else {
+                if (idx === 0) {
```

Comment:
is the array always the same way? Its never structured differently? One thing that worries me about this if else statement is that if the structure ever changes it would break. 

Is there another check that we can make that is a bit more stringent?